### PR TITLE
[FEATURE] Add get data context mercury v1 integration test.

### PR DESCRIPTION
**Pull request by billdirks**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/cloud/end_to_end/test_datacontext_v1_api.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    context = gx.get_context(
+        mode="cloud",
+        cloud_base_url=os.environ.get("GX_CLOUD_BASE_URL"),
+        cloud_organization_id=os.environ.get("GX_CLOUD_ORGANIZATION_ID"),
+        cloud_access_token=os.environ.get("GX_CLOUD_ACCESS_TOKEN"),
+    )
+    assert isinstance(context, CloudDataContext)
+    # This assert is to ensure we are hitting the v1 and not the v0 endpoint.
+    # We do this by asserting the config version is above anything that will appear in v0.
+    # assert context.config_version >= 4.0
```
Comment (billdirks) [inline tests/integration/cloud/end_to_end/test_datacontext_v1_api.py]: I will uncomment this assert once the mercury stub change goes in that updates the config version number.

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 2]
```diff
@@ -328,13 +338,16 @@ def _change_value_from_v0_to_v1(config: dict, key: str, v0_value: str, v1_value:
         return config
 
     @classmethod
-    def _request_cloud_backend(cls, cloud_config: GXCloudConfig, uri: str) -> Response:
```
Comment (tyler-hoffman) [inline great_expectations/data_context/data_context/cloud_data_context.py:331]: Thanks, this was a bit confusing!

File: great_expectations/data_context/store/gx_cloud_store_backend.py [hunk 3]
```diff
@@ -624,14 +624,18 @@ def construct_versioned_url(
         """Construct the correct url for a given resource."""
         version = cls._ENDPOINT_VERSION_LOOKUP.get(resource_name, EndpointVersion.V0)
 
+        # urljoin strips the last component of the path in base_url if it doesn't end
+        # with a / which we don't want so we add a / to the end if one isn't present
+        b = base_url if base_url[-1] == "/" else base_url + "/"
```
Comment (tyler-hoffman) [inline great_expectations/data_context/store/gx_cloud_store_backend.py]: Non-blocking: should we be sanitizing this at the config layer?

Also even though it's a short-lived variable, can we give it a descriptive name? `sanitized_base_url` or something?

**Review by tyler-hoffman — APPROVED** [review]
Approved, pending renaming `b` 😄

File: great_expectations/data_context/store/gx_cloud_store_backend.py (hunk 3 above)
Comment (billdirks) [reply great_expectations/data_context/store/gx_cloud_store_backend.py]: You are right about doing this at the config layer. I'll track that down and either update this PR or fix in a follow up.

I'll change the name of this variable :)

Comment (billdirks) [reply great_expectations/data_context/store/gx_cloud_store_backend.py]: I've moved this sanitization into the `__init__` of `GXCloudConfig`.

Comment (tyler-hoffman) [reply great_expectations/data_context/store/gx_cloud_store_backend.py]: Nice! thanks!

**Pull request merged**